packages/cli/commands/theme/marketplace-validate.js

```diff
@@ -35,67 +32,116 @@ exports.handler = async options => {
   await loadAndValidateOptions(options);
 
   const accountId = getAccountId(options);
-  const absoluteSrcPath = path.resolve(getCwd(), src);
-  let stats;
+
+  trackCommandUsage('validate', null, accountId);
+
+  const spinnies = new Spinnies();
+
+  spinnies.add('marketplaceValidation', {
+    text: i18n(`${i18nKey}.logs.validatingTheme`, {
+      path: src,
+    }),
+  });
+
+  // Kick off validation
+  let requestResult;
+  const assetType = 'THEME';
+  const requestGroup = 'EXTERNAL_DEVELOPER';
   try {
-    stats = fs.statSync(absoluteSrcPath);
-    if (!stats.isDirectory()) {
-      logger.error(
-        i18n(`${i18nKey}.errors.invalidPath`, {
-          path: src,
-        })
-      );
-      return;
-    }
-  } catch (e) {
-    logger.error(
-      i18n(`${i18nKey}.errors.invalidPath`, {
-        path: src,
-      })
-    );
-    return;
+    requestResult = await requestValidation(accountId, {
+      path: src,
+      assetType,
+      requestGroup,
+    });
+  } catch (err) {
+    logger.debug(err);
+    process.exit(EXIT_CODES.ERROR);
   }
 
-  if (!options.json) {
-    logger.log(
-      i18n(`${i18nKey}.logs.validatingTheme`, {
-        path: src,
-      })
-    );
+  // Poll till validation is finished
+  try {
+    const checkValidationStatus = async () => {
+      const validationStatus = await getValidationStatus(accountId, {
+        validationId: requestResult,
+      });
+
+      if (validationStatus === 'REQUESTED') {
+        await new Promise(resolve => setTimeout(resolve, SLEEP_TIME));
+        await checkValidationStatus();
+      }
+    };
+
+    await checkValidationStatus();
+
+    spinnies.remove('marketplaceValidation');
+  } catch (err) {
+    logger.debug(err);
+    process.exit(EXIT_CODES.ERROR);
   }
-  trackCommandUsage('validate', null, accountId);
 
-  const themeFiles = await walk(absoluteSrcPath);
-
-  applyAbsoluteValidators(
-    MARKETPLACE_VALIDATORS.theme,
-    absoluteSrcPath,
-    themeFiles,
-    accountId
-  ).then(groupedResults => {
-    logValidatorResults(groupedResults, { logAsJson: options.json });
-
-    if (
-      groupedResults
-        .flat()
-        .some(result => result.result === VALIDATION_RESULT.FATAL)
-    ) {
-      process.exit(EXIT_CODES.WARNING);
+  // Fetch the validation results
+  let validationResults;
+  try {
+    validationResults = await getValidationResults(accountId, {
+      validationId: requestResult,
+    });
+  } catch (err) {
+    logger.debug(err);
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  if (validationResults.errors.length) {
+    const { errors } = validationResults;
+
+    errors.forEach(err => {
+      logger.error(`${err.context}`);
+
+      if (err.failureReasonType === 'DOWNLOAD_EMPTY') {
+        logger.log(
```

Comment:
I think this extra log is causing us to log a duplicate error in the case where the theme is not found. Example:
<img width="741" alt="image" src="https://user-images.githubusercontent.com/6654014/198663547-f40f9f7a-68df-4f0c-b33d-0d99d47a37eb.png">

Reply:
Maybe just rendering the top "No asset exists at path..." log is enough?

Reply:
I can remove that now 👍